File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let frequency = {};
    for(let num of nums){
        frequency[num] = (frequency[num] || 0) +1;
    }
    let max=0;
    let maxVal = 0;

    for (const[key,value] of Object.entries(frequency)){
        if(value>max){
            max = value;
            maxVal = Number(key);
        }
    }
    return maxVal;

};

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> frequency;
        for (int num : nums) {
        frequency[num]++;
    }
    int max=0;
    int maxVal = 0;
     for (const auto& pair : frequency) {
        if(pair.second>max){
        max= pair.second;
        maxVal = pair.first;
        }
    }
    return maxVal;
    }
};